documentation/tutorial.md

```diff
@@ -255,18 +306,31 @@ just makes an `empty run`. Let’s integrate the extraction logic above
 into our spider.
 
 ```elixir
-defmodule Homebase do
+defmodule BooksToScrape do
   use Crawly.Spider
 
   @impl Crawly.Spider
-  def base_url(), do: "https://www.homebase.co.uk"
+  def base_url(), do: "https://books.toscrape.com/catalogue/category/books/"
 
   @impl Crawly.Spider
   def init() do
+    # Get all the cateory URLs to use as our starting point
+    response = Crawly.fetch("https://books.toscrape.com/catalogue/category/books_1/index.html")
+
+    {:ok, document} = Floki.parse_document(response.body)
+
+    # Extract product category URLs
+    product_categories_urls =
```

Comment:
category parsing can be in the parse_item, 

parse_item is flexible enough for multi pages structure extraction, dont need to specially have these in the init.

plus having it in the init makes it more complicated and may cause the spider to crash on error

Reply:
So there's product categories on each product category page, that would mean every time parse item runs, it'll attempt to add those urls again, although, it wouldn't go through. Just thought there wasn't any need to deliberately attempt to add the same urls multiple times and relying on the fact that they'll be parsed only once.